Packs/PrismaCloudCompute/Integrations/PaloAltoNetworks_PrismaCloudCompute/PaloAltoNetworks_PrismaCloudCompute.py

```diff
@@ -253,9 +271,701 @@ def fetch_incidents(client):
+    Args:
+        date_string (str): the date in string representation.
+        new_format (str): the new requested format for the date string.
+
+    Returns:
+        str: date as a new format, in case of a failure returns the original date string.
+    """
+    try:
+        return dateparser.parse(date_string=date_string).strftime(new_format)
+    except AttributeError:
+        return date_string
+
+
+def epochs_to_timestamp(epochs: int, date_format: str = "%B %d, %Y %H:%M:%S %p") -> str:
+    """
+    Converts epochs time representation to a new string date format.
+
+    Args:
+        epochs (int): time in epochs (seconds)
+        date_format (str): the desired format that the timestamp will be.
+
+    Returns:
+        str: timestamp in the new format, empty string in case of a failure
+    """
+    try:
+        return datetime.utcfromtimestamp(epochs).strftime(date_format)
+    except TypeError:
+        return ""
+
+
+def get_api_filtered_response(
+    client: PrismaCloudComputeClient,
+    url_suffix: str,
+    offset: int,
+    limit: int,
+    args: Optional[dict] = None,
+) -> list:
+    """
+    Filter the api response according to the offset/limit, used in case the api doesn't support limit/offset
+
+    Args:
+        client (PrismaCloudComputeClient): prisma-cloud-compute client.
+        url_suffix (str): url suffix of the base api url.
+        offset (int): the offset from which to begin listing the response.
+        limit (int): the maximum limit of records in the response to fetch.
+        args (dict): any command arguments if exist.
+
+    Returns:
+        list: api filtered response, empty list in case there aren't any records in the api response.
+    """
+    if not args:
+        args = {}
+
+    response = client.api_request(method='GET', url_suffix=url_suffix, params=assign_params(**args))
+
+    if response:
+        start = min(offset, len(response))
+        end = min(offset + limit, len(response))
+        return response[start:end]
+
+    return []
+
+
+def build_single_host_profile_table_response(host_info: dict) -> str:
+    """
+    Build a table for a single host.
+
+    Args:
+        host_info (dict): host information from the api.
+
+    Returns:
+        str: markdown table output for a single host.
+    """
+    host_description_table = build_hostnames_description_table(
+        host_description_info=get_hostname_description_info(host_info=host_info)
+    )
+
+    apps_info = [
+        {
+            'HostId': host_info.get('_id'),
+            'AppName': app.get('name'),
+            'StartupProcess': app.get('startupProcess').get('path'),
+            'User': app.get('startupProcess').get('user'),
+            'LaunchTime': parse_date_string_format(date_string=app.get('startupProcess').get('time'))
+        } for app in host_info.get('apps', [])
+    ]
+    ssh_events_info = [
+        {
+            'User': event.get('user'),
+            'Ip': str(ipaddress.IPv4Address(event.get('ip'))),
```

Comment:
please add comment here why this is needed

Reply:
ack